contrib/IECoreArnold/src/IECoreArnold/ParameterAlgo.cpp

```diff
@@ -125,9 +125,20 @@ void setParameterInternal( AtNode *node, const char *name, int parameterType, bo
 				}
 				break;
 			case AI_TYPE_ENUM :
-				if( const StringData *data = dataCast<StringData>( name, value ) )
+				// Arnold supports setting enums with either the integer index or the string name
+				if( runTimeCast<const StringData>( value ) )
 				{
-					AiNodeSetStr( node, name, data->readable().c_str() );
+					if( const StringData *data = dataCast<StringData>( name, value ) )
+					{
+						AiNodeSetStr( node, name, data->readable().c_str() );
+					}
+				}
+				else
```

Comment:
If this was `else if( ...` we could save some indentation and make things a little more readable. Perhaps we should also reverse things so that the string test is second, so when passing neither a string nor an int, we get a `dataCast` warning that StringData is expected? I think that is still probably our preferred method of specifying an enum value?